### ISSUE TITLE
Hash to group must be domain-separated.

### ISSUE THREAD

**Issue by armfazh** [drand/kyber#50]
The current interface for hashing to group elements must consider the use of domain separation strings.

This is aligned with hash to curve IETF specification. All hashes must be domain-separated, so a protocol can specify the separation when the hash is invoked as different random oracles. See more https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-16#name-domain-separation-requireme

Current:
https://github.com/drand/kyber/blob/5706fa5694922bff1603f5d87e88cdd56da1873d/hash.go#L13-L15

Proposed:

```go
type HashablePoint interface { 
    Hash(msg, dst []byte) Point 
} 
```

**Comment by ineiti** [comment]
I don't think this is kyber's task to do. Kyber is the "shoot in your foot" crypto library. 

If you want to have kyber for actual applications for people who don't know what they're doing, there is a _lot_ to do. 

And if you start nudging it here and there in a more fool-proof version, you might anger researchers who won't find their required primitives anymore.

I propose to close this issue and keep kyber as "shoot in your foot" library. Other libraries can build on top of kyber to have a more user-friendly way of handling crypto.
